src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -713,29 +712,32 @@ protected void removeDoneFiles(List<String> dataFileNames) {
         }
     }
 
-    /**
-     * Builds a document description for New or Edit PAAT documents generated by this class.
-     * Will prefix the description with payee's principal name.
-     */
-    protected String getDocumentDescription(Person payee, boolean isEntityId, boolean isNewAccount) {
+    protected String buildDocumentDescription(Person payee, PayeeACHAccount achAccount, MaintenanceDocument document) {
+        boolean isNewAccount = isDocumentCreatingNewAccount(document);
+        boolean isEntityId = StringUtils.equalsIgnoreCase(PayeeIdTypeCodes.ENTITY, achAccount.getPayeeIdentifierTypeCode());
         int descMaxLength = dataDictionaryService.getAttributeMaxLength(DocumentHeader.class.getName(), KRADPropertyConstants.DOCUMENT_DESCRIPTION).intValue();
         StringBuilder docDescription = new StringBuilder(descMaxLength);
         
         docDescription.append(payee.getPrincipalName());
         docDescription.append(isNewAccount ? " -- New " : " -- Edit ");
-        docDescription.append(isEntityId ? "Entity" : "Employee").append(" Account");
+        docDescription.append(isEntityId ? "Entity" : "Employee");
+        docDescription.append(" Account");
         
-        return (descMaxLength < docDescription.length()) ? docDescription.substring(0, descMaxLength) : docDescription.toString();
+        return StringUtils.left(docDescription.toString(), descMaxLength);
+    }
+
+    protected boolean isDocumentCreatingNewAccount(MaintenanceDocument document) {
+        return StringUtils.equalsIgnoreCase(KFSConstants.MAINTENANCE_NEW_ACTION, document.getNewMaintainableObject().getMaintenanceAction());
     }
 
     /**
-     * LOG error and throw RunTimeException
+     * LOG error and throw RuntimeException
      * 
      * @param errorMessage
      */
-    // Copied this method from CustomerLoadServiceImpl.
+    // Copied and tweaked this method from CustomerLoadServiceImpl.

```

Comment:
is this comment needed?

Reply:
No, the comment is not really needed; it must have been left over from before we started becoming stricter about comment usage. I will remove it.